### PR TITLE
Add COCOBboxDataset

### PR DESCRIPTION
**Pull request by yuyu2172**
Depends on #452

I think we need some modification to the BboxDataset API.

1. bbox can be zero-sized `(0,4)`.
2. Make it easier to filter images with no ground truth bounding boxes. I suggest to make a public method `labels` for all bbox datasets. The method returns a list of bounding boxes. Also, I suggest to support similar method for all label datasets.

### CODE REVIEW

**Comment by Hakuyume** [comment]
>  Make it easier to filter images with no ground truth bounding boxes. I suggest to make a public method labels for all bbox datasets. The method returns a list of bounding boxes. Also, I suggest to support similar method for all label datasets.

I guess the advantage of `labels` is that users can filter images without loading them. Is ii right?

**Comment by yuyu2172** [comment]
Yea

**Comment by Hakuyume** [comment]
I made a mix-in for this problem. Please check #454.

**Comment by yuyu2172** [comment]
Areas of instance masks are needed for evaluation. So, I added an option that makes the dataset to return them.

**Comment by yuyu2172** [comment]
ok

File: docs/source/reference/datasets.rst [hunk 1]
```diff
@@ -51,6 +51,13 @@ CityscapesTestImageDataset
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 .. autoclass:: CityscapesTestImageDataset
 
+COCO
```
Comment (Hakuyume) [inline docs/source/reference/datasets.rst]: I prefer `MS COCO` because we use `PASCAL VOC` in this doc.

**Pull request merged**